### DISCUSSION TITLE
Does the calculation require biomarkers to be input with specific units

### DISCUSSION THREAD

**Discussion by 91489puth** [dayoonkwon/BioAge#35] [Q&A]
Thank you for the package and guidance; it has been very helpful. We selected six biomarkers (albumin, ALP, creatinine, glucose, WBC, and lymphocytes) to calculate biological age. However, we have encountered some issues:

```
> hd = hd_nhanes(biomarkers=c("albumin","alp","creat","glucose","wbc","lymph"))
> kdm = kdm_nhanes(biomarkers=c("albumin","alp","creat","glucose","wbc","lymph"))
> phenoage = phenoage_nhanes(biomarkers=c("albumin_gL","alp","creat_umol","glucose_mmol","wbc","lymph"))
```

1. Does the calculation require biomarkers to be input with specific units, such as albumin in g/L, creatinine in µmol/L, and glucose in mmol/L for the calculation of PhenoAge?

2. Must the input format be consistent across different BioAge calculations? Specifically, is it correct to use albumin, ALP, creatinine, glucose, WBC, and lymphocytes for HD and KDM calculations, while using albumin_gL, ALP, creat_umol, glucose_mmol, WBC, and lymphocytes for PhenoAge?

3. Are transformations necessary for non-normally distributed markers? For example, should we use creatinine or its natural logarithm (lncreat)?

4. In our new dataset, the HD values range from 0.09 to 12.01, KDM values range from -21 to 527, and PhenoAge values range from 8.7 to 124.2. Are these ranges typical for biological age measures? We are unsure about the expected ranges for these calculations.

Any guidance on these points would be greatly appreciated.

**Comment by dayoonkwon** [comment]
1. Yes. PhenoAge uses `albumin_gL`, `creat_umol`, and `glucose_mmol` because the original Levine algorithm was trained on those units. The variable names tell you which units to use.

2. Yes, that's correct. HD and KDM use standard US units (`albumin` in g/dL, `creat` in mg/dL, `glucose` in mg/dL), while PhenoAge uses SI-converted units. This is by design — each algorithm needs to match the units used in its original training.

3. The `_nhanes` functions handle this internally. For `_calc` functions, use whatever form was used in training. Log-transformed versions (`lncreat`, `lncrp`, etc.) are pre-computed in the NHANES datasets.

4. KDM values ranging from -21 to 527 indicate a problem — likely a unit mismatch in your input data. Expected KDM values should be roughly in the range of chronological age (e.g., 20–90). I'd double-check that your biomarker units match the NHANES conventions. See `vignettes/NHANES_Codebook.pdf` for the full unit reference.
